tools/autograd/templates/VariableType.cpp

```diff
@@ -45,6 +47,11 @@ namespace{
 }
 
 namespace {
+
+bool torch_dispatch_set() {
+  return static_cast<bool>(at::impl::TorchDispatchModeTLS::get_state());
```

Comment:
This would be more general if we also test for Python key on the argument tensors as well, as the assert disabling here only works for modes.

Reply:
Just to confirm we agree on this. These assert are failing for a valid reason here right? The view semantic is not properly implemented by this mode (view of an outside Tensor will not be properly updated if an inplace happens inside). So this disable is only intended to unblock experimentation while we work on a fix for that?

Reply:
I definitely agree that they are very restrictive and can be annoying when trying things out. But that doesn't make them wrong. And for sure not a good reason to just disable them.
Maybe we want to add an env variable to disable them to ease local development?